File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> numSet = new HashSet<>();
        for(int n:nums) {
            numSet.add(n);
        }

        int longestStreak = 0;
        for(int num:nums){
            if(!numSet.contains(num-1)) {
                int length = 1;
                while(numSet.contains(num + length)){
                    length++;
                }
                longestStreak = Math.max(length, longestStreak);
            }
        }
        return longestStreak;
    }
}

File: Data Structures & Algorithms/house-robber/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    rob(nums) {
        //[rob1, rob2, n, n+1, ...]
        let rob1 = 0;
        let rob2 = 0;
        for(let num of nums) {
            let temp = Math.max(num + rob1, rob2);
            rob1 = rob2;
            rob2 = temp;
        }
        return rob2
    }
}
